internal/resources/aws/dynamodb_table.go

```diff
@@ -23,29 +26,69 @@ type DynamoDbTableArguments struct {
 	MonthlyStreamsReadRequestUnits *int64 `json:"monthlyStreamsReadRequestUnits,omitempty"`
 }
 
-func (args *DynamoDbTableArguments) PopulateUsage(u *schema.UsageData) {
-	if u != nil {
-		args.MonthlyWriteRequestUnits = u.GetInt("monthly_write_request_units")
-		args.MonthlyReadRequestUnits = u.GetInt("monthly_read_request_units")
-		args.StorageGB = u.GetInt("storage_gb")
-		args.PitrBackupStorageGB = u.GetInt("pitr_backup_storage_gb")
-		args.OnDemandBackupStorageGB = u.GetInt("on_demand_backup_storage_gb")
-		args.MonthlyDataRestoredGB = u.GetInt("monthly_data_restored_gb")
-		args.MonthlyStreamsReadRequestUnits = u.GetInt("monthly_streams_read_request_units")
+func (args *DynamoDbTableArguments) buildUsageSchema(keysToSkipSync []string) []*schema.UsageSchemaItem {
+	schema := []*schema.UsageSchemaItem{
+		{Key: "region", DefaultValue: "us-east-1", ValueType: schema.String},
+		{Key: "billing_mode", DefaultValue: "PROVISIONED", ValueType: schema.String},
+		{Key: "replica_regions", DefaultValue: []string{}, ValueType: schema.StringArray},
+		{Key: "write_capacity", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "read_capacity", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "monthly_write_request_units", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "monthly_read_request_units", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "pitr_backup_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "on_demand_backup_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "monthly_data_restored_gb", DefaultValue: 0, ValueType: schema.Int64},
+		{Key: "monthly_streams_read_request_units", DefaultValue: 0, ValueType: schema.Int64},
 	}
+	setShouldSync(schema, keysToSkipSync)
+	return schema
 }
 
-var DynamoDbTableUsageSchema = []*schema.UsageSchemaItem{
-	{Key: "monthly_write_request_units", DefaultValue: 0, ValueType: schema.Int64},
-	{Key: "monthly_read_request_units", DefaultValue: 0, ValueType: schema.Int64},
-	{Key: "storage_gb", DefaultValue: 0, ValueType: schema.Int64},
-	{Key: "pitr_backup_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
-	{Key: "on_demand_backup_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
-	{Key: "monthly_data_restored_gb", DefaultValue: 0, ValueType: schema.Int64},
-	{Key: "monthly_streams_read_request_units", DefaultValue: 0, ValueType: schema.Int64},
+func (args *DynamoDbTableArguments) populateFromUsage(u *schema.UsageData) {
```

Comment:
The idea is to map values from usage to args.  Note that usage take precedence over existings values -- the idea is to give the end user a way to override the IaC parsing.  I guess this is readable but it seems very verbose.  I'm going to look into putting "`infracost:`" tags on DynamoDbTableArguments.